src/OpenTK.Windowing.GraphicsLibraryFramework/JoystickState.cs

```diff
@@ -0,0 +1,197 @@
+//
+// JoystickState.cs
+//
+// Copyright (C) OpenTK
+//
+// This software may be modified and distributed under the terms
+// of the MIT license. See the LICENSE file for details.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Text;
+using OpenTK.Windowing.Common.Input;
+
+[assembly: InternalsVisibleTo("OpenTK.Windowing.Desktop")]
+
+namespace OpenTK.Windowing.GraphicsLibraryFramework
+{
+    /// <inheritdoc />
+    internal sealed class JoystickState : IJoystickState
```

Comment:
I really dislike `sealed` classes, who are we to decide what the user can and cannot do...
(and because this is internal there is really no need for it to be `sealed` either)

Reply:
The further issue is what someone would do with a custom joystick implementation. They can't inject it into NativeWindow.  If they need any further functionality, they'd probably be best wrapping an IJoystickState or writing extension methods if they really had to.

We're into the territory of writing code to drive behaviours here. Many devs will do stuff because it's there, and that's not what we generally want them to do. We wouldn't want someone to get an instance of JoystickState and call Update() on it themselves, for example. Since we're going for immutability, this is important.

In any case, if someone did want a very specific implementation of JoystickState, OpenTK is open source and they can just fork it.

Reply:
Generally by keeping things sealed it forces API developers to create proper extension points using interfaces.

It also serves as a form of self-documentation to indicate "there are no subclasses of this type to worry about".